cmd/cueckoo/cmd/importpr.go

```diff
@@ -48,28 +55,84 @@ func importPRDef(c *Command, args []string) error {
 		return fmt.Errorf("%q is not a valid number", pr)
 	}
 
-	// TODO: validate that the number provided is indeed a valid PR
+	log.Printf("using github remote URL %q", cfg.githubURL)
 
-	// TODO: work out support for multiple PRs
+	branchName := fmt.Sprintf("importpr-%d", pr)
 
-	msg := fmt.Sprintf("Import PR %d from GitHub to Gerrit", pr)
-	payload, err := buildImportPRPayload(msg, importPRPayload{
-		PR: pr,
-	})
-	if err != nil {
+	// TODO: Note that mainErr's ctx is not wired here.
+	// We should wire it up and use it for e.g. FetchContext.
+	// For now, use a hard-coded timeout of 10s.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// If the branch already exists, refuse to continue.
+	if out, err := run(ctx,
+		"git", "show-ref", "--verify", "--quiet", fmt.Sprintf("refs/heads/%s", branchName),
+	); err == nil {
+		return fmt.Errorf("branch %q already exists; exiting. delete it to start over.", branchName)
+	} else if err != nil && len(out) > 0 {
```

Comment:
Why the check on `len(out) > 0` here?

Reply:
because a non-zero exit status with zero output means that the branch doesn't exist and nothing went wrong. do you think I should clarify that?

Reply:
Doh. No, that makes total sense. Slow start to the day :)

Reply:
Rewrote this to be clearer and added a comment.